test/e2e/upgrades/serviceaccount_admission_controller_migration.go

```diff
@@ -0,0 +1,141 @@
+// and pod-after-migration are able to make requests using in cluster config.
+func (t *ServiceAccountAdmissionControllerMigrationTest) Test(f *framework.Framework, done <-chan struct{}, upgrade UpgradeType) {
+	ginkgo.By("Waiting for upgrade to finish")
+	<-done
+
+	ginkgo.By("Starting post-upgrade check")
+	ginkgo.By("Checking pod-before-migration makes successful requests using in cluster config")
+	podBeforeMigration, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), podBeforeMigrationName, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	if podBeforeMigration.GetUID() != t.pod.GetUID() {
+		framework.Failf("Pod %q GetUID() = %q, want %q.", podBeforeMigration.Name, podBeforeMigration.GetUID(), t.pod.GetUID())
+	}
+	if podBeforeMigration.Status.ContainerStatuses[0].RestartCount != 0 {
+		framework.Failf("Pod %q RestartCount = %d, want 0.", podBeforeMigration.Name, podBeforeMigration.Status.ContainerStatuses[0].RestartCount)
+	}
+	if len(podBeforeMigration.Spec.Volumes) != 1 || podBeforeMigration.Spec.Volumes[0].Secret == nil {
+		framework.Failf("Pod %q Volumes[0].Secret = nil, want non-nil.", podBeforeMigration.Name)
+	}
+	inClusterClientMustWork(f, podBeforeMigration)
+
+	ginkgo.By("Checking pod-after-migration makes successful requests using in cluster config")
+	podAfterMigration := createPod(f, podAfterMigrationName)
+	if len(podAfterMigration.Spec.Volumes) != 1 || podAfterMigration.Spec.Volumes[0].Projected == nil {
+		framework.Failf("Pod %q Volumes[0].Projected.Sources = nil, want non-nil.", podAfterMigration.Name)
+	}
+	inClusterClientMustWork(f, podAfterMigration)
+
+	ginkgo.By("Finishing post-upgrade check")
+}
+
+// Teardown cleans up any remaining resources.
+func (t *ServiceAccountAdmissionControllerMigrationTest) Teardown(f *framework.Framework) {
+	// rely on the namespace deletion to clean up everything
+}
+
+func inClusterClientMustWork(f *framework.Framework, pod *v1.Pod) {
+	var logs string
+	since := time.Now()
+	if err := wait.PollImmediate(15*time.Second, 5*time.Minute, func() (done bool, err error) {
+		framework.Logf("Polling logs")
+		logs, err = e2epod.GetPodLogsSince(f.ClientSet, pod.Namespace, pod.Name, "inclusterclient", since)
+		if err != nil {
+			framework.Logf("Error pulling logs: %v", err)
+			return false, nil
+		}
+		numTokens, err := e2eauth.ParseInClusterClientLogs(logs)
+		if err != nil {
+			return false, fmt.Errorf("inclusterclient reported an error: %v", err)
+		}
+		if numTokens == 0 {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		framework.Failf("Unexpected error: %v\n%s", err, logs)
+	}
+}
+
+// createPod creates a pod.
+func createPod(f *framework.Framework, podName string) *v1.Pod {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name:  "inclusterclient",
+				Image: imageutils.GetE2EImage(imageutils.Agnhost),
```

Comment:
add a shorter poll period with `"--poll-interval=..."` (like 5 or 10... default is 30) so we don't have to wait so long to pick up logs

Reply:
good point, done.